packages/flutter/test/material/tabs_test.dart

```diff
@@ -2677,6 +2691,142 @@ void main() {
     // There was a time where this would throw an exception
     // because we tried to send a notification on dispose.
   });
+
+  testWidgets('TabController animation should not get interrupted when user drags TabBarView', (WidgetTester tester) async {
+    const List<Tab> tabs = <Tab>[
+      Tab(text: 'A'), Tab(text: 'B'), Tab(text: 'C')
+    ];
+    TabController tabController;
+
+    Widget buildTabControllerFrame(BuildContext context, TabController controller) {
+      tabController = controller;
+      return MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(
+            bottom: TabBar(
+              controller: controller,
+              tabs: tabs,
+            ),
+          ),
+          body: TabBarView(
+            controller: controller,
+            children: tabs.map((Tab tab) {
+              return Center(child: Text(tab.text));
+            }).toList(),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(TabControllerFrame(
+      builder: buildTabControllerFrame,
+      length: tabs.length,
+      initialIndex: 0,
+    ));
+
+    expect(tabController.index, 0);
+    tabController.animateTo(2, duration: const Duration(seconds: 1), curve: Curves.linear);
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 500));
+
+    // Drag the TabBarView while tabController is in animation
+    // and check if it didn't get interrupted
+    final TestGesture gesture = await tester.startGesture(tester.getCenter(find.byType(PageView)));
+    await gesture.moveBy(const Offset(1.0, 0.0));
+    expect(tabController.animation.value, 1.0);
+    expect(tabController.indexIsChanging, true);
+
+    await tester.pumpAndSettle();
```

Comment:
Is this needed?

Reply:
Oh I thought ```tester.pumpAndSettle()``` is needed to really end the test by finishing all the animations going on. Isn't this needed?

Reply:
It is ok to end the test in the middle of animation unless there are something specific you want to test.

Reply:
Done!